### ISSUE TITLE
 AsyncRemover thread stuck

### ISSUE THREAD

**Issue by iAlaska** [Alluxio/alluxio#12444]
**Alluxio Version:**
We use both 2.2.0 and 2.3.0

**Describe the bug**
Below is  our use scenario：
1.Mount some hdfs directory (contains time based partitions) on alluxio
2.Maintain a time window that every day, Load newest partition into alluxio and Free oldest partition out of alluxio

After some Maintain cycle，all of AsyncRemover thread stuck at acquiring ClientRWLock
```
"block-removal-service-9" #2184 daemon prio=5 os_prio=0 tid=0x00007fe336e9d000 nid=0x340399 waiting on condition [0x00007fdae7cfb000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x000000065d50c5b8> (a java.util.concurrent.Semaphore$NonfairSync)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireShared(AbstractQueuedSynchronizer.java:967)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireShared(AbstractQueuedSynchronizer.java:1283)
        at java.util.concurrent.Semaphore.acquireUninterruptibly(Semaphore.java:494)
        at alluxio.worker.block.ClientRWLock$SessionLock.lock(ClientRWLock.java:92)
        at alluxio.worker.block.BlockLockManager.lockBlock(BlockLockManager.java:114)
        at alluxio.worker.block.TieredBlockStore.removeBlock(TieredBlockStore.java:378)
        at alluxio.worker.block.TieredBlockStore.removeBlock(TieredBlockStore.java:371)
        at alluxio.worker.block.DefaultBlockWorker.removeBlock(DefaultBlockWorker.java:477)
        at alluxio.worker.block.AsyncBlockRemover$BlockRemover.run(AsyncBlockRemover.java:106)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
 
   Locked ownable synchronizers:
        - <0x0000000600919d38> (a java.util.concurrent.ThreadPoolExecutor$Worker)
```

I do some investigate on heap, found that
- Size of AsyncRemover thread pool is 10
- `SessionLock` instances is greater 10 than `LockRecord`
```
 num     #instances         #bytes  class name
----------------------------------------------
...
 91:          1234          39488  alluxio.worker.block.BlockLockManager$LockRecord
 109:          1244          29856  alluxio.worker.block.ClientRWLock$SessionLock
```
Maybe there are some cases that releases of LockRecord is out of control?

**To Reproduce**
Unfortunately i can't exactly reproduce this bug, but it happens frequently.

**Expected behavior**
AsyncRemover should release the lock anyway to ensure that next remove loop not stuck.

**Urgency**
The only way that can clear the un-removed blocks is that, remove all Worker processed and re-execute Free command. But this will cause user query fail, so it is very expensive to recover.

**Additional context**
Maybe related to this issue: https://github.com/Alluxio/alluxio/issues/11309

**Comment by apc999** [comment]
@maobaolong  let's take a look at this issue together. If this is too hard or requires too much code change, I will re-assign this to other core maintainers.

**Comment by maobaolong** [comment]
@apc999 Ok, this issue seems urgent, bring bad user experience, will take a look at this issue.

**Comment by LuQQiu** [comment]
@maobaolong 
@apc999 and I discussed some of the potential fixes
1) catch throwable -> done in your #12475 
2) Release timeout locks, asyncLockRemover
3) use lock.tryLock() instead of lock.lock()
Thanks

**Comment by maobaolong** [comment]
@LuQQiu Thanks for the review on #12475 .
I think the 1 and 3 are in the  #12475 , about 2, we can create a new pr, and it will be better if we can provide a view of top time hold lock, or is it possible to release the locks by administrator?

**Comment by apc999** [comment]
shall be solved by #12643
